src/audio/copier.c

```diff
@@ -539,9 +539,53 @@ static int copier_reset(struct comp_dev *dev)
 	return ret;
 }
 
+/* visit connected  pipeline to find the dai comp and latency.
+ * Latency means the difference of processed sample bytes
+ * between host and dai component.
+ */
+static struct comp_dev *find_dai_comp(int pipeline_id, int *latency)
```

Comment:
This should be in pipeline logic to prove a common method to get the bulk delay (latency in ms from to to DAI presentation). I think we already have some methods to get the DAI which can be reused. The copier and other processing could then use the common logic.

Reply:
@lgirdwood I only found ipc_get_ppl_sink_comp for single pipeline and didn't find a function to search connected pipeline list. Do you know which function support this ? Thanks!